hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionHealth.java

```diff
@@ -0,0 +1,70 @@
+package pl.allegro.tech.hermes.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Objects;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Sets.immutableEnumSet;
+import static java.util.Collections.emptySet;
+
+public final class SubscriptionHealth {
+    public static final SubscriptionHealth HEALTHY = new SubscriptionHealth(Status.HEALTHY, emptySet());
+    public static final SubscriptionHealth NO_DATA = new SubscriptionHealth(Status.NO_DATA, emptySet());
+
+    private final Status status;
+    private final ImmutableSet<SubscriptionHealthProblem> problems;
```

Comment:
Yeah, but actually, this *is* implementation (a private field). :) The getter returns a Set. I've left ImmutableSet here just for the sake of clarity that SubscriptionHealth is immutable.